### PR TITLE
Univariate time series using Recurrent Neural Networks

### LINKED ISSUE #7197: [Machine Learning] Univariate time series using Recurrent Neural Networks
#### Proposed title of article
[Machine Learning] Univariate time series using Recurrent Neural Networks

#### Proposed article introduction
A recurrent neural network (RNN) is a class of artificial neural network where connections between nodes form a direct graph along a sequence. This allows it to exhibit temporal dynamic behavior for a time sequence. 
Unlike feedforward neural networks, RNNs can use their internal state (memory) to process sequences of inputs. This makes them applicable to price prediction but also handwriting and speech recognition.

LSTM are a variant of RNN(recurrent neural network) and are widely used of for time series projects in forecasting and future predictions. Long short-term memory (LSTM) units are units of a recurrent neural network (RNN). LSTM networks are well-suited to classifying, processing and making predictions based on time-series data.

In time series, we have univariate, bivariate, and multivariate modeling. Univariate statistics summarize only one variable at a time, bivariate statistics compare two variables and multivariate statistics compare more than two variables. 
In this tutorial, we will be focusing on univariate time series modeling. Time series forecasting is the process of analyzing time series data using statistics and modeling to make predictions and inform strategic decision-making. Time series analysis involves developing models to gain an understanding of the data to understand the underlying causes.

We will use RNN to build a time series model that predicts monthly milk production.

#### Key takeaways 
1. What are RNNs? 
2. Types of RNNs and Improvements to RNN 
3. Dataset preparation.
4. Time series analysis and decomposition.
5. Time series prediction with only one variable using RNN.

#### Article quality
This article is unique because we will discuss all the types of RNNs and improvements to RNN in details. This will give a reader a good understanding of the RNN architecture. We will explain all the concepts involved in time series so that a beginner can easily follow.

We will then perform time series decompositon so that a reader can know te components that make up a time series. We will also diferentiate univariate, bivariate, and multivariate modeling in clear way. 
Finally we will implement a custom time serie model that predicts monthly milk production. The tutorial wil have a good flow of steps that a reader can easily follow.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by jamesmainamaingi9**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #7197

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete

File: content/articles/univariate-time-series-using-recurrent-neural-networks/index.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Prerequisites]
+- A reader should have the basic knowledge and understanding of the following before implementing the time series model:
+
+- Some basic understanding and knowledge of [time series terms and concepts](/engineering-education/introduction-to-time-series/)
+- A brief introduction to [neural networks architecture and concepts](/engineering-education/introduction-to-neural-networks/)
+- Know how to build an [artificial neural network with Keras](/engineering-education/build-ann-with-keras/)
+- How to run a Python project in [Google Colab](https://colab.research.google.com/)
```
Comment (Bayler) [inline content/articles/univariate-time-series-using-recurrent-neural-networks/index.md:48]:
```suggestion
- [Time series terms and concepts](/engineering-education/introduction-to-time-series/)
- A brief introduction to [neural networks architecture and concepts](/engineering-education/introduction-to-neural-networks/)
- Know how to build an [artificial neural network with Keras](/engineering-education/build-ann-with-keras/)
- How to run a Python project in [Google Colab](https://colab.research.google.com/)
```

File: content/articles/univariate-time-series-using-recurrent-neural-networks/index.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+
+Recurrent neural networks use internal memory that helps them to store information retrieved from the previous data points and use this information to generate the next data points.
+
+Recurrent neural networks implement a feedback loop within the layers with the hidden layer which distinguishes them from the traditional artificial neural networks.
+
+The image below shows the distinction between a traditional artificial neural network and a recurrent neural network:
+
+![Traditional neural network vs RNN ](/engineering-education/univariate-time-series-using-recurrent-neural-networks/ann-vs-rnn.png)
+
+[Image Source: Simplilearn](https://www.simplilearn.com/ice9/free_resources_article_thumb/Simple_Recurrent_Neural_Network.png)
```
Comment (Bayler) [inline content/articles/univariate-time-series-using-recurrent-neural-networks/index.md:61]:
```suggestion
![Traditional neural network vs RNN ](/engineering-education/univariate-time-series-using-recurrent-neural-networks/ann-vs-rnn.png)

*[Image Source: Simplilearn](https://www.simplilearn.com/ice9/free_resources_article_thumb/Simple_Recurrent_Neural_Network.png)*
```

Comment (Bayler) [reply content/articles/univariate-time-series-using-recurrent-neural-networks/index.md:61]: This is the correct format for adding an image from other sources. Please fix the others too accordingly.

**Comment by Bayler** [comment]
@jamesmainamaingi9 In addition to the above revisions, please pass your article through Grammarly premium and fix those grammatical errors.

**Comment by jamesmainamaingi9** [comment]
Hello @Bayler, I have worked on all the requested changes.

**Comment by Bayler** [comment]
@jamesmainamaingi9 Welcome to our Eng-Ed Community. This being your first tutorial, I'm impressed with your technical writing ability. However, there is always room for improvement. Take note of all the commits made in this tutorial to improve on your writing skill. This PR will now proceed for final review.

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**